lib/web_ui/lib/src/engine/skwasm/skwasm_impl/filters.dart

```diff
@@ -60,16 +62,20 @@ class SkwasmImageFilter implements SceneImageFilter {
     return SkwasmImageFilter._(imageFilterCompose(nativeOuter.handle, nativeInner.handle));
   }
 
-  void dispose() {
-    if (!_isDisposed) {
-      imageFilterDispose(handle);
-      _isDisposed = true;
-    }
-  }
+  static final SkwasmFinalizationRegistry<RawImageFilter> _registry =
+    SkwasmFinalizationRegistry<RawImageFilter>(imageFilterDispose);
 
+  @override
   final ImageFilterHandle handle;
   bool _isDisposed = false;
 
+  void dispose() {
```

Comment:
Drive-by-comment: this code is duplicated a LOT – worth making it a `mixin` or something?

Reply:
I took a couple different passes at it trying to DRY this out, this is actually more concise than it was in my previous passes. I did try actually factoring this out into a mixin, but there were two things that made it difficult:

(1) Each object has a different cleanup function for its handle, and since a mixin can't have a generative constructor it's hard for the adopting class to pass it in.
(2) The finalization registry is a static on the derived class, so the derived class can access it. I can't put the registry on the mixin because you'd only get one finalization registry for the mixin and not one per derived class.

There are probably clever ways to avoid these issues, but at a certain point I stopped refactoring and figured this is good enough. If you feel strongly, I can go back and try to figure out a way to fold some of this logic into `SkwasmObjectWrapper` though... maybe...

Reply:
Looking at the code, it's clear we'd need some funky new language features to generalize this. Nevermind. 😢

Reply:
`SkwasmFinalizationRegistry` already takes a reference to the type-specific clean-up function.  Maybe this is doable via `SkwasmFinalizationRegistry`. (see https://github.com/flutter/engine/pull/42328#discussion_r1206056640)